src/traces/parcoords/plot.js

```diff
@@ -103,7 +119,17 @@ module.exports = function plot(gd, cdparcoords) {
                 gdDimensions[i].splice(gdDimensionsOriginalOrder[i].indexOf(d), 0, d); // insert at original index
             });
 
-        gd.emit('plotly_restyle');
+        // TODO: we can't really store this part of the interaction state
+        // directly as below, since it incudes data arrays. If we want to
+        // persist column order we may have to do something special for this
+        // case to just store the order itself.
+        // Registry.call('_storeDirectGUIEdit',
+        //     gd.data[inputIndices[i]],
+        //     fullLayout._tracePreGUI[gd._fullData[fullIndices[i]]._fullInput.uid],
+        //     {dimensions: gdDimensions[i]}
+        // );
```

Comment:
Feels like the better way to handle this is to create a `dimensionorder` attribute, so we can just store that, rather than directly mutating the `dimensions` array.

Like `columnorder` in `table` traces, but [that feature isn't fully built out either](https://github.com/plotly/plotly.js/pull/3236/commits/a11ec44f388741dd21046d08a782e4e76fc8ba22#diff-1e0dea3a9361823fe80cbbdc50befd7eR591) so I left both of these as open items for now.

Reply:
This probably deserves a new issue :rocket: